src/algebra/category/Module/abelian.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2020 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import algebra.category.Module.kernels
+import algebra.category.Module.limits
+import category_theory.abelian.basic
+
+/-!
+# The category of left R-modules is abelian.
+-/
+
+open category_theory
+open category_theory.limits
+
+noncomputable theory
+
+universe u
+
+namespace Module
+variables {R : Type u} [ring R] {M N : Module R} (f : M ⟶ N)
+
+/-- In the category of modules, every monomorphism is normal. -/
+def normal_mono [mono f] : normal_mono f :=
+{ Z := of R f.range.quotient,
+  g := f.range.mkq,
+  w := linear_map.range_mkq_comp _,
+  is_limit :=
+    is_kernel.iso_kernel _ _ (kernel_is_limit _)
+      /- The following invalid Lean code might help you understand what's going on here:
```

Comment:
@gebner @digama0 How hard would it be to make the claim that this is invalid Lean code an invalid claim?

Reply:
Not too hard I think. [Relevant zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/homeomorph.2Etrans.20and.20calc)